src/union_room.c

```diff
@@ -0,0 +1,4607 @@
+        {
+            data->state = 13;
+        }
+        else if (gReceivedRemoteLinkPlayers != 0)
+        {
+            sub_80FAFE0(1);
+            data->state++;
+        }
+        break;
+    case 16:
+        DestroyListMenuTask(data->listTaskId, 0, 0);
+        CopyBgTilemapBufferToVram(0);
+        RemoveWindow(data->listWindowId);
+        DestroyTask(data->field_17);
+        Free(data->field_8);
+        Free(data->field_0);
+        Free(data->field_4);
+        sub_800AB9C();
+        data->state++;
+        break;
+    case 17:
+        if (IsLinkTaskFinished())
+            DestroyTask(taskId);
+        break;
+    }
+}
+
+void MEvent_CreateTask_CardOrNewsWithFriend(u32 arg0)
+{
+    u8 taskId;
+    struct UnkStruct_Group * dataPtr;
+
+    taskId = CreateTask(sub_8117F20, 0);
+    sUnionRoomMain.group = dataPtr = (void*)(gTasks[taskId].data);
+    sGroup = dataPtr;
+
+    dataPtr->state = 0;
+    dataPtr->textState = 0;
+    dataPtr->field_12 = arg0 - 21;
+    gSpecialVar_Result = 0;
+}
+
+static void sub_8117F20(u8 taskId)
+{
+    s32 id;
+    struct WindowTemplate winTemplate1, winTemplate2;
+    struct UnkStruct_Group * data = sUnionRoomMain.group;
+
+    switch (data->state)
+    {
+    case 0:
+        SetHostRFUtgtGname(data->field_12 + 21, 0, 0);
+        sub_800B1F4();
+        OpenLink();
+        sub_80FBBD8();
+        data->field_4 = AllocZeroed(0x70);
+        data->field_0 = AllocZeroed(0x200);
+        data->state = 1;
+        break;
+    case 1:
+        AddTextPrinterToWindow1(gUnknown_8458FE4);
+        data->state = 2;
+        break;
+    case 2:
+        sub_811A650(data->field_4, 4);
+        sub_811A5E4(data->field_0->arr, 16);
+        data->field_11 = sub_811A054(data->field_4, data->field_12 + 7);
+
+        winTemplate1 = gUnknown_8456D4C;
+        winTemplate1.baseBlock = GetMysteryGiftBaseBlock();
+        data->listWindowId = AddWindow(&winTemplate1);
+
+        data->field_D = AddWindow(&gUnknown_8456D54);
+
+        MG_DrawTextBorder(data->listWindowId);
+        gMultiuseListMenuTemplate = gUnknown_8456DDC;
+        gMultiuseListMenuTemplate.windowId = data->listWindowId;
+        data->listTaskId = ListMenuInit(&gMultiuseListMenuTemplate, 0, 0);
+
+        MG_DrawTextBorder(data->field_D);
+        FillWindowPixelBuffer(data->field_D, PIXEL_FILL(1));
+        PutWindowTilemap(data->field_D);
+        sub_8115924(data->field_D);
+        CopyWindowToVram(data->field_D, 2);
+
+        CopyBgTilemapBufferToVram(0);
+        data->field_F = 0;
+        data->state = 3;
+        break;
+    case 3:
+        id = sub_8116FE4();
+        switch (id)
+        {
+        case 1:
+            PlaySE(SE_PC_LOGIN);
+        default:
+            RedrawListMenu(data->listTaskId);
+            break;
+        case 0:
+            id = ListMenu_ProcessInput(data->listTaskId);
+            if (JOY_NEW(A_BUTTON) && id != -1)
+            {
+                // this unused variable along wit
```

Comment:
does `MALE` `FEMALE` apply here?

Reply:
Which line was this meant to be on? GitHub borked it up

Reply:
2541